lib/messageformat.js

```diff
@@ -304,13 +304,14 @@ MessageFormat.prototype._precompile = function(ast, data) {
       return '{ ' + r.join(', ') + ' }';
 
     case 'string':
-      tmp = '"' + (ast.val || "").replace(/\n/g, '\\n').replace(/"/g, '\\"') + '"';
+      tmp = JSON.stringify(ast.val || "");
       if ( data.pf_count ) {
         args = [ propname(data.keys[data.pf_count-1], 'd') ];
         if (data.offset[data.pf_count-1]) args.push(data.offset[data.pf_count-1]);
-        tmp = tmp.replace(/(^|[^\\])#/g, '$1"+' + 'number(' + args.join(', ') + ')+"');
+        tmp = tmp.replace(/(^|[^\\\\])#/g, '$1"+' + 'number(' + args.join(', ') + ')+"');
```

Comment:
Yes, you are completely right. I wanted to change line 315 and ended up editing the wrong line.